src/run/ongoing.ts

```diff
@@ -13,19 +14,34 @@ export interface FinishedRun {
   assertions: Assertion[];
 }
 
+const MAX_CONSECUTIVE_ERRORS = 5;
+
 export const waitForRunEnd = async (client: ApiClient, startedRun: StartedRun): Promise<FinishedRun> => {
-  let runInfo: RunInformationResponse;
+  let runInfo: RunInformationResponse | undefined;
```

Comment:
It can be undefined? xD

Reply:
It's undefined until the first successful call. We then either exit the loop with this variable defined, of by throwing an exception, so it's always available when we use it to build the return value (which wouldn't compile otherwise, BTW).